src/main/java/com/minecolonies/coremod/entity/ai/citizen/herders/AbstractEntityAIHerder.java

```diff
@@ -0,0 +1,534 @@
+     *
+     * @param job the job to fulfill.
+     * @param maxAnimalsMultiplier the number of max animals to add per hut level
+     */
+    public AbstractEntityAIHerder(@NotNull final J job, final int maxAnimalsMultiplier)
+    {
+        super(job);
+        this.maxAnimalMultiplier = maxAnimalsMultiplier;
+        super.registerTargets(
+          new AITarget(IDLE, START_WORKING),
+          new AITarget(START_WORKING, this::startWorkingAtOwnBuilding),
+          new AITarget(PREPARING, this::prepareForHerding),
+          new AITarget(HERDER_DECIDE, this::decideWhatToDo),
+          new AITarget(HERDER_BREED, this::breedAnimals),
+          new AITarget(HERDER_BUTCHER, this::butcherAnimals),
+          new AITarget(HERDER_PICKUP, this::pickupItems)
+        );
+        worker.setCanPickUpLoot(true);
+    }
+
+    @Override
+    protected int getActionsDoneUntilDumping()
+    {
+        return ACTIONS_FOR_DUMP;
+    }
+
+    @NotNull
+    @Override
+    protected List<ItemStack> itemsNiceToHave()
+    {
+        final List<ItemStack> list = super.itemsNiceToHave();
+        list.add(new ItemStack(getBreedingItem(), 2));
+        return list;
+    }
+
+    /**
+     * Decides what job the herder should switch to, breeding or Butchering.
+     *
+     * @return The next {@link AIState} the herder should switch to, after executing this method.
+     */
+    public AIState decideWhatToDo()
+    {
+        setDelay(DELAY_FOURTY);
+
+        final List<T> animals = new ArrayList<>(searchForAnimals());
+
+        if (animals.isEmpty())
+        {
+            setDelay(DELAY_ONE_HUNDRED);
+            return HERDER_DECIDE;
+        }
+
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.herder.deciding"));
+
+        final int numOfBreedableAnimals = animals.stream().filter(animal -> animal.getGrowingAge() == 0).toArray().length;
+
+        final boolean hasBreedingItem = worker.hasItemInInventory(getBreedingItem(), 0);
+
+        if (!searchForItemsInArea().isEmpty())
+        {
+            return HERDER_PICKUP;
+        }
+        else if (maxAnimals())
+        {
+            return HERDER_BUTCHER;
+        }
+        else if (numOfBreedableAnimals >= NUM_OF_ANIMALS_TO_BREED && hasBreedingItem)
+        {
+            return HERDER_BREED;
+        }
+        return START_WORKING;
+    }
+
+    /**
+     * Redirects the herder to their building.
+     *
+     * @return The next {@link AIState}.
+     */
+    private AIState startWorkingAtOwnBuilding()
+    {
+        worker.setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.herder.goingToHut"));
+        if (walkToBuilding())
+        {
+            return getState();
+        }
+        return PREPARING;
+    }
+
+    /**
+     * Prepares the herder for herding
+     *
+     * @return The next {@link AIState}.
+     */
+    private AIState prepareForHerding()
+    {
+        toolsNeeded.add(ToolType.AXE);
```

Comment:
why do you add it to the lists? Why not call the methods that first check if you actually need to request things?